src/views/DeleteData.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <h3>Delete Data</h3>
+
+  <div class="container">
+    {{ welcomeMessage }}
+    <table style="width: 100%">
+      <tr>
+        <th>User ID</th>
+        <th>Title</th>
+        <th>Data</th>
+        <th>Gene</th>
+        <th>Date</th>
+        <th>Delete</th>
+      </tr>
+
+      <tr v-for="chart in chartList" :key="chart.chartOptions">
+        <td>
+          {{ chart[1].user_id }}
+        </td>
+        <td>
+          {{ chart[1].chart_title }}
+        </td>
+        <td>
+          {{ chart[1].data_source }}
+        </td>
+        <td>
+          {{ chart[1].gene }}
+        </td>
+        <td>
+          {{ chart[1].date_created }}
+        </td>
+        <td>
+          <div class="form-field mx-auto">
+            <button
+              v-on:click="DeleteData(chart[0])"
+              class="btn-large red lighten-2"
+            >
+              X
+            </button>
+          </div>
+        </td>
+      </tr>
+    </table>
+  </div>
+</template>
+
+<script>
+import { ref } from "vue";
+import { firebaseDb, firebaseAuth } from "../utils/firebase";
+
+export default {
+  mounted: function () {
+    this.$nextTick(this.GetAllUserCharts);
+  },
```

Comment:
This method calls the getAllUserCharts method as soon as the page loads basically